aether-ui/aether/ui/api/utils.py

```diff
@@ -171,212 +169,147 @@ def kernel_data_request(url='', method='get', data=None):
-    else:
-        pipeline.kernel_refs[object_name] = res['id']
-        pipeline.save()
-    return pipeline
-
-
-def update_kernel_object(object_name, id, data):
-    return kernel_data_request(f'{object_name.lower()}s/{id}/', 'put', data)
-
-
 def is_object_linked(kernel_refs, object_name, entity_type_name=''):
     if kernel_refs and object_name in kernel_refs:
         try:
-            if object_name is 'schema' or object_name is 'projectSchema':
+            if object_name is 'schemas':
                 if entity_type_name in kernel_refs[object_name]:
-                    url = f'{object_name.lower()}s/{kernel_refs[object_name][entity_type_name]}'
+                    url = f'{object_name.lower()}/{kernel_refs[object_name][entity_type_name]}'
                 else:
                     return False
             else:
-                url = f'{object_name.lower()}s/{kernel_refs[object_name]}/'
-            kernel_data_request(url, 'get')
+                url = f'{object_name.lower()}/{kernel_refs[object_name]}/'
+            kernel_data_request(url)
             return True
         except Exception:
             return False
     else:
         return False
 
 
-def create_project_schema_object(name, pipeline, schema_id, entity_name):
-    if not is_object_linked(pipeline.kernel_refs, 'projectSchema', entity_name):
-        project_schema_data = {
-                            'name': name,
-                            'mandatory_fields': '[]',
-                            'transport_rule': '[]',
-                            'masked_fields': '[]',
-                            'is_encrypted': False,
-                            'project': pipeline.kernel_refs['project'],
-                            'schema': schema_id
-                        }
-        create_new_kernel_object('projectSchema',
-                                 pipeline, project_schema_data, entity_name=entity_name)
-    return pipeline
-
-
-def publish_preflight(pipeline, project_name, outcome):
+def publish_preflight(pipeline, outcome, contract):
     '''
     Performs a check for possible pipeline publish errors against kernel
     '''
-    if pipeline.mapping_errors:
-        outcome['error'].append('Mappings have errors')
+    if contract.mapping_errors:
+        outcome['error'].append('{} mappings have errors'.format(contract.name))
         return outcome
-    for entity_type in pipeline.entity_types:
-        if is_object_linked(pipeline.kernel_refs, 'schema', entity_type['name']):
+    for entity_type in contract.entity_types:
+        if is_object_linked(contract.kernel_refs, 'schemas', entity_type['name']):
             outcome['exists'].append({entity_type['name']: '{} schema with id {} exists'.format(
                                         entity_type['name'],
-                                        pipeline.kernel_refs['schema'][entity_type['name']])})
+                                        contract.kernel_refs['schemas'][entity_type['name']])})
```

Comment:
another styling comment... lines with several indentations are hard to follow.

```python
outcome['exists'].append({
    entity_type['name']: '{} schema with id {} exists'.format(
        entity_type['name'],
        contract.kernel_refs['schemas'][entity_type['name']]),
})
```

As an example...